app/components/nextflow/samplesheet/column_component.rb

```diff
@@ -38,21 +38,41 @@ def render_cell_type(property, entry, sample, fields, index) # rubocop:disable M
       end
 
       def render_fastq_cell(sample, property, entry, fields, index)
-        direction = property.match(/fastq_(\d+)/)[1].to_i == 1 ? :pe_forward : :pe_reverse
-        files = sample.sorted_files[direction] || []
-        data = {}
-        if files.empty? && property != 'fastq_2'
-          files = sample.sorted_files[:singles] || []
+        direction = get_fastq_direction(property)
+        files = get_fastq_files(entry, sample, direction, pe_only: property['pe_only'].present?)
+        data = get_fastq_data(files, direction, index, property)
+        render_file_cell(property, entry, fields, files, @required, data, files&.first)
+      end
+
+      private
+
+      def get_fastq_direction(property)
+        property.match(/fastq_(\d+)/)[1].to_i == 1 ? :pe_forward : :pe_reverse
+      end
+
+      def get_fastq_files(_entry, sample, direction, pe_only: false)
+        singles = sample.sorted_files[:singles].filter do |file|
```

Comment:
**nitpick (blocking):** Any reason why you are not using `filter_filtes_by_pattern` method to perform the filtering?

Reply:
This is also raising an exception when I select a sample that has no files attached to it.

![image](https://github.com/user-attachments/assets/f96033b2-25f9-4d64-84b0-f98bc02f2816)